commandhelper/commandhelper.go

```diff
@@ -0,0 +1,75 @@
+package commandhelper
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	"github.com/bitrise-io/go-steputils/output"
+	"github.com/bitrise-io/go-utils/colorstring"
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/log"
+)
+
+// RawOutputCouldNotBeExporterError ...
+type RawOutputCouldNotBeExporterError struct {
+	wrappedError error
+}
+
+func (e RawOutputCouldNotBeExporterError) Error() string {
+	return fmt.Sprintf("could not export output: %s", e.wrappedError)
+}
+
+// RunAndExportOutputWithReturningLastNLines ...
+func RunAndExportOutputWithReturningLastNLines(cmd *command.Model, destinationPath, envKey string, lines int) (string, error) {
+	var outBuffer bytes.Buffer
+	preStdout := cmd.GetCmd().Stdout
+	preStderr := cmd.GetCmd().Stderr
+	cmd.SetStdout(&outBuffer)
+	cmd.SetStderr(&outBuffer)
+
+	defer func() {
```

Comment:
I would create a new instance of the input argument cmd and set the stds on that so we no need to take responsibility about the mutation to the outer scope

Reply:
because this function we exactly want to do not use any stdout/stderr but return N lines and export file

Reply:
I will rather use cmd as a struct instead of a pointer.